### PR TITLE
crypto: add progress info to the redux store

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers CC @thebearjew 

This PR adds a listener for crypto operation progress info, and stores it as `state.crypto.{decrypt,encrypt,sign,verify}.bytes{Complete,Total}` in Redux.

* rename `keybase1.OperationType` to `keybase1.SaltpackOperationType`
* add listeners for saltpack{Start,Progress,Done} -- currently ignoring Start and Done in the reducer, but still firing their actions.
* hook up the reducer to save the payload.

### CODE REVIEW

**Comment by thebearjew** [comment]
@cjb awesome! Will take a look in 5 hours >.<

File: shared/reducers/crypto.tsx [hunk 1]
```diff
@@ -27,6 +27,8 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
 
     if (operationGuard(operation, action)) return
 
+    draftState[operation].bytesComplete = 0
+    draftState[operation].bytesTotal = 0
```
Comment (thebearjew) [inline shared/reducers/crypto.tsx]: @cjb What happens if the service continues to push progress after the input is cleared?

E.g. Encrypting an `X`Gb file and then clear the input (effectively canceling) before the file operation is complete?

Comment (cjb) [reply shared/reducers/crypto.tsx]: Dunno! We could remove the zeroing from clearRecipients if it doesn't work well.  We don't have good ordering guarantees (e.g. we get Progress after Done sometimes) so perhaps just displaying the latest Progress output all the time is going to look best..

Comment (patrickxb) [reply shared/reducers/crypto.tsx]: If the ordering is a problem, I can make the notifications more synchronous.

Comment (thebearjew) [reply shared/reducers/crypto.tsx]: Hmm, perhaps cancelling a file operation is something I didn’t fully consider. Especially for large files.

Regardless of the progress notifications being synchronous, it seems that halting an saltpack operation on a file is impossible currently.

**Comment by cjb** [comment]
merging despite many CI flakes

**Pull request merged**